Quadriga/src/main/java/edu/asu/spring/quadriga/rest/ConceptCollectionRestController.java

```diff
@@ -448,5 +465,60 @@
         }
 
     }
+    
+    
+    /**
+     * Rest interface to get conceptcollections related to project
+     * http://<<URL>:<PORT>>/quadriga/auth/rest/<projectid>/conceptcollections.json
+     * http://localhost:8080/quadriga/auth/rest/PROJ_bb7ad41b-3e85-4309-b2ff-47d644307b9b/conceptcollections.json
+     * 
+     * 
+     * @author Ajay Modi & Bharath Srikantan
+     * @param projectid
+     * @param model
+     * @param principal
+     * @return
+     * @throws QuadrigaStorageException
+     * @throws QuadrigaAccessException
+     */
 
+    @RequestMapping(value = "auth/rest/{projectid}/conceptcollections.json", method = RequestMethod.GET, produces = "application/json")
+    public ResponseEntity<String> listProjectConceptCollectionJson(
+            @PathVariable("projectid") String projectid, Model model,
+            Principal principal) throws QuadrigaStorageException,
+            QuadrigaException {
+        String userId = principal.getName();
+        List<IProjectConceptCollection> projectConceptCollectionList = null;
+        try {
+            // TODO: listProjectConceptCollection() needs to be modified
+            projectConceptCollectionList = projectConceptCollectionManager
+                    .listProjectConceptCollection(projectid, userId);
+            
+            JSONArray ja = new JSONArray();
+            
+            if(projectConceptCollectionList!=null){
+                for (IProjectConceptCollection conceptCollection : projectConceptCollectionList) {
+                    JSONObject j = new JSONObject();
+                    try {
+
+                        j.put("id", conceptCollection.getConceptCollection()
+                                .getConceptCollectionId());
+                        j.put("name", conceptCollection.getConceptCollection()
+                                .getConceptCollectionName());
+                        ja.put(j);
+
+                    } catch (JSONException e) {
+                        // TODO Auto-generated catch block
+                        throw new QuadrigaException(e.getMessage(), e);
+                    }
+
+                }
+            }
+            
+            return new ResponseEntity<String>(ja.toString(), HttpStatus.OK);
+            
+        } catch (QuadrigaStorageException e) {
+            throw new QuadrigaStorageException();

```

Comment:
the try/catch block should be where the exception is actually thrown. If one collection can't be retrieved but the others can, you want to show the the once that you could retrieve and just show an error. Also, given that this is a method that returns JSON, return proper error codes and useful error messages. And don't swallow the exception ;)